main/chanvars.c

```diff
@@ -52,6 +52,17 @@ struct ast_var_t *_ast_var_assign(const char *name, const char *value, const cha
 	return var;
 }
 
+int ast_var_set(const struct ast_var_t *var, const char *new_value)
+{
+	if (var->value) {
+		ast_free(var->value);
+	}
+
```

Comment:
This doesn't seem right... you're potentially freeing `var->value` so it's no longer valid, then trying to copy a string to that location?

Reply:
Moot now. ast_var_set() has been removed